src/components/Planning/ProjectProgrammedDialog/ProjectProgrammedSearch.tsx

```diff
@@ -0,0 +1,168 @@
+import { getProjectsWithParams } from '@/services/projectServices';
+import { arrayHasValue } from '@/utils/common';
+import { SearchInput } from 'hds-react/components/SearchInput';
+import { FC, memo, useCallback, useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useOptions } from '@/hooks/useOptions';
+import { IProgrammedProjectSuggestions, ISearchRequest } from '@/interfaces/searchInterfaces';
+import { IClass } from '@/interfaces/classInterfaces';
+import { useAppSelector } from '@/hooks/common';
+import {
+  selectAllClasses,
+  selectSelectedClass,
+  selectSelectedSubClass,
+} from '@/reducers/classSlice';
+import SelectedProjectCard from './SelectedProjectCard';
+
+interface ISearchState {
+  searchWord: string;
+  searchedProjects: Array<IProgrammedProjectSuggestions>;
+}
+interface IProjectSearchProps {
+  onProjectClick: (value: IProgrammedProjectSuggestions | undefined) => void;
+  projectsForSubmit: Array<IProgrammedProjectSuggestions>;
+  onProjectSelectionDelete: (projectName: string) => void;
+}
+
+const buildBreadCrumbs = (path: string, classes: Array<IClass>): Array<string> =>
+  path.split('/').map((p) => classes.find((c) => c.id === p)?.name || '');
+
+const ProjectProgrammedSearch: FC<IProjectSearchProps> = ({
+  onProjectClick,
+  projectsForSubmit,
+  onProjectSelectionDelete,
+}) => {
+  const scrollRef = useRef<HTMLDivElement>(null);
+  const classes = useAppSelector(selectAllClasses);
+
+  const [searchState, setSearchState] = useState<ISearchState>({
+    searchWord: '',
+    searchedProjects: [],
+  });
+  const { searchedProjects, searchWord } = searchState;
+  const phaseProposal =
+    useOptions('phases', true).find((phase) => phase.label === 'proposal')?.value || '';
+  const phaseDesign =
+    useOptions('phases', true).find((phase) => phase.label === 'design')?.value || '';
+  const selectedClass = useAppSelector(selectSelectedClass);
+  const selectedSubClass = useAppSelector(selectSelectedSubClass);
+
+  useEffect(() => {
```

Comment:
some comments here would give some information what is this for

Reply:
done